src/core/lender/distributor/distributor.sol

```diff
@@ -44,36 +37,42 @@ contract Distributor is DSNote {
     function deny(address usr) public auth note { wards[usr] = 0; }
     modifier auth { require(wards[msg.sender] == 1); _; }
 
-    ManagerLike public manager;
+    address public shelf;
 
-    constructor(address manager_)  public {
+    // --- Tranches ---
+    address public senior;
+    address public junior;
+
+    constructor(address shelf_)  public {
         wards[msg.sender] = 1;
-        manager = ManagerLike(manager_);
+        shelf = shelf_;
     }
 
     function depend (bytes32 what, address addr) public auth {
-        if (what == "manager") { manager = ManagerLike(addr); }
+        if (what == "shelf") { shelf = addr; }
+        if (what == "junior") { junior = addr; }
+        else if (what == "senior") { senior = addr;}
         else revert();
     }
 
     function balance() public {
-        (uint action, uint currencyAmount) = manager.requestAction();
+        (uint action, uint currencyAmount) = ShelfLike(shelf).requestAction();
```

Comment:
I don't like the method name: `requestAction` since it gives you an amount as well.